### ISSUE TITLE
Serial connections are fine but ubootwrite.py crashes out and I cannot upload the mr33-uboot.bin file Help Appreciated

### ISSUE THREAD

**Issue by antonical** [riptidewave93/LEDE-MR33#16]
HI I have managed to get all of the various components together for flashing the MR33's

I have 2 out of 10 that have the correct firmware the rest all have the 2017 firmware.

When I run the python program. I get the following the instant I connect power to the device.

./ubootwrite.py` --write=mr33-uboot.bin
Uploading image
Traceback (most recent call last):
  File "./ubootwrite.py", line 221, in <module>
    main()
  File "./ubootwrite.py", line 216, in main
    upload(ser, options.write, int(options.size, 0), int(options.addr, 0), options.verbose, debug, options.shell)
  File "./ubootwrite.py", line 170, in upload
    ret = memwrite(ser, path, size, start_addr, verbose, debug, shell)
  File "./ubootwrite.py", line 93, in memwrite
    prompt = getprompt(ser, start_addr, verbose, shell)
  File "./ubootwrite.py", line 36, in getprompt
    buf = ser.read(256);
  File "/usr/local/lib/python2.7/dist-packages/serial/serialposix.py", line 501, in read
    'device reports readiness to read but returned no data '
serial.serialutil.SerialException: device reports readiness to read but returned no data (device disconnected or multiple access on port?)

I can confirm that I have tested the serial connection and all seems well in a terminal and all seems well. I have also tried exiting the serial port terminal to ensure it is not holding onto the connection and also removed and reinserted the adaptor. 

i get the same error every time.

Can you assist with troubleshooting please?

Cheers
Tony

**Comment by antonical** [comment]
Just adding I can talk to the device fine using the adaptor and can get a prompt that starts:
STINKBUG # 

But the .py program using the same adaptor just crashes out as above. I can't really find anything useful on google to try and understand the error or how to resolve it. The serial port terminal has the facility to write a raw file to the device. Not sure how I can progress with getting this up and running.

Cheers
Tony

**Comment by antonical** [comment]
I managed to sort this out it was caused by the 3.3V pin of my adaptor on pin 1 of the 4 pin header. As soon as I removed it and ran with the tx rx and gnd pins all was well and I could upload the mr33-uboot.bin file.

I went with the latest snapshot initramfs and the 19.07.3 release.

All went well and I now have a very fast and usable openwrt AP and WWAN repeater.

Thank you for all your hard work in getting this process to work. Hopefully there will be an answer for the later firmware as these are seriously capable units and look very slick. 

Cheers
Tony
